15_tictactoe/go/grid/grid_test.go

```diff
@@ -0,0 +1,160 @@
+package grid
+
+import (
+	"slices"
+	"testing"
+)
+
+func TestString(t *testing.T) {
+	testGrid := Grid{
+		Cells: [9]int{1, 1, 1, 0, 2, 1, 0, 1, 1},
+	}
+	test := testGrid.String()
+	wantedResult := `
++---+---+---+
+| 1 | 1 | 1 |
++---+---+---+
+| 0 | 2 | 1 |
++---+---+---+
+| 0 | 1 | 1 |
++---+---+---+`
+	if test != wantedResult {
+
+		t.Errorf("testGrid.String() = %v; want %v", test, wantedResult)
+	}
+}
+
+func TestIsWin(t *testing.T) {
+	tests := []struct {
+		g     Grid
+		wantIsWin bool
+		wantWinner string
+	}{
+		{
+			g: Grid{
+				Cells: [9]int{1, 1, 1, 0, 2, 1, 0, 1, 1},
+			},
+			wantIsWin: true,
+			wantWinner: "X",
+		},
+		{
+			g: Grid{
+				Cells: [9]int{0, 0, 0, 0, 0, 0, 0, 0, 0},
+			},
+			wantIsWin: false,
+			wantWinner: ""},
+	}
+
+	for _, test := range tests {
+		got1, got2 := test.g.IsWin()
+		if got1 != test.wantIsWin && got2 != test.wantWinner {
+			t.Errorf("IsWin(%v) = %v, %v, want %v and %v", test.g, got1, got2, test.wantIsWin, test.wantWinner)
+		}
+	}
+}
+
+func TestFreeCells(t *testing.T) {
+	// tests := []struct {
+	// 	g    Grid
+	// 	want []int
+	// }{
+	// 	{
+	// 		g: Grid{
+	// 			Cells: [9]int{1, 1, 1, 0, 2, 1, 0, 1, 1},
+	// 		},
+	// 		want: []int{3, 6},
+	// 	},
+	// 	{
+	// 		g: Grid{
+	// 			Cells: [9]int{0, 0, 0, 0, 0, 0, 0, 0, 0},
+	// 		},
+	// 		want: []int{0, 1, 2, 3, 4, 5, 6, 7, 8},
+	// 	},
+	// }
+
+	// testGrid struct to test multiple cases
+	testGrid := []struct {
+		g Grid
```

Comment:
Try to do this instead:
```suggestion
		g Grid
		wanted []int
```

Reply:
I think I have done it!?